source/localizable/tutorial/autocomplete-component.md

```diff
@@ -1,101 +1,170 @@
-The `key-up` property will be bound to the `autoComplete` action,
-passed in to the component from the `index` controller. The `autoComplete`
-action takes the `filter` property as the argument when invoked.
+Since we plan to use ember-data as our model store, we need to make our action calls to fetch data asynchronous, so we'll return promises.  Because we are async, we want to use the wait helper at the end of our test, which will wait for all promises to resolve before completing the test.
 
-The template also contains a button that is bound to the `search` action.
-Similar to the `autoComplete` action, the `search` action is passed in from
-the `index` controller and takes the `filter` property when invoked.
-
-Lastly, the `filter-listing.hbs` template contains an unordered list,
-that displays the `city` property of each item in the `filteredList`
-property in our component. Clicking the list item will fire the `choose`
-action with the `city` property of the item as a parameter, which will
-then populate the `input` field with the name of that `city`.
+```tests/integration/components/filter-listing-test.js
+import Ember from 'ember';
+import { moduleForComponent, test } from 'ember-qunit';
+import hbs from 'htmlbars-inline-precompile';
+import wait from 'ember-test-helpers/wait';
 
-Here is what the component's JavaScript looks like:
+moduleForComponent('filter-listing', 'Integration | Component | filter listing', {
+  integration: true
+});
 
-```app/components/filter-listing.js
-export default Ember.Component.extend({
-  filter: null,
-  filteredList: null,
+const ITEMS = [{city: 'San Francisco'}, {city: 'Portland'}, {city: 'Seattle'}];
+const FILTERED_ITEMS = [{city: 'San Francisco'}];
 
-  actions: {
-    choose(city) {
-      this.set('filter', city);
+test('should initially load all listings', function (assert) {
+  // we want our actions to return promises, since they are potentially fetching data asynchronously
+  this.on('filterByCity', (val) => {
+    if (val === '') {
+      return Ember.RSVP.resolve(ITEMS);
+    } else {
+      return Ember.RSVP.resolve(FILTERED_ITEMS);
     }
-  }
+  });
+  
+  // with an integration test, you can set up and use your component in the same way your application 
+  // will use it.
+  this.render(hbs`
+    {{#filter-listing filter=(action 'filterByCity') as |rentals|}}
+      <ul>
+      {{#each rentals as |item|}}
+        <li class="city">
+          {{item.city}}
+        </li>
+      {{/each}}
+      </ul>
+    {{/filter-listing}}
+  `);
+
+  // the wait function will return a promise that will wait for all promises 
+  // and xhr requests to resolve before running the contents of the then block.
+  return wait().then(() => {
+    assert.equal(this.$('.city').length, 3);
+    assert.equal(this.$('.city').first().text().trim(), 'San Francisco');
+  });
 });
 ```
+For our second test we'll check that the component will actually reduce the listings shown based on the result of the filter action.
```

Comment:
I feel like this could be simplified, "we'll check the filter action actually filters the listings shown"?